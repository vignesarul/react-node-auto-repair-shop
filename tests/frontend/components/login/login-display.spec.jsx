jest.mock('react-router-dom/Link');
import React from 'react';
import { mount } from 'enzyme';
import LoginDisplay from 'components/login/login-display';

describe('<LoginDisplay />', () => {
  let wrapper;
  const props = {
    history: {
      push: jest.fn(),
    },
    isLoading: false,
    performLogin: jest.fn(),
  };

  beforeEach(() => {
    wrapper = mount(<LoginDisplay {...props} />);
  });

  it('should match the snapshop', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
