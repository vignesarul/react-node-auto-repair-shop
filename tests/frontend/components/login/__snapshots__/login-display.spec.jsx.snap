// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LoginDisplay /> should match the snapshop 1`] = `
<Login
  error={Object {}}
  history={
    Object {
      "push": [Function],
    }
  }
  info=""
  isLoading={false}
  performLogin={[Function]}
  user={Object {}}
>
  <div
    className="py-5"
  >
    <div
      className="container"
    >
      <div
        className="row mx-auto w-100"
      >
        <div
          className="col-12 col-lg-5 col-xl-5 mx-auto col-md-8 align-self-center"
        >
          <div
            className="card mx-auto w-100"
          >
            <div
              className="card-header"
            >
              Login
            </div>
            <AlertMessage
              message={
                Object {
                  "error": Object {},
                  "history": Object {
                    "push": [Function],
                  },
                  "info": "",
                  "isLoading": false,
                  "performLogin": [Function],
                  "user": Object {},
                }
              }
            >
              <div />
            </AlertMessage>
            <div
              className="card-block"
            >
              <form
                onSubmit={[Function]}
              >
                <div
                  className="form-group"
                >
                  <label
                    htmlFor="email"
                  >
                    Email address
                  </label>
                  <input
                    className="form-control"
                    name="email"
                    required={true}
                    type="email"
                  />
                </div>
                <div
                  className="form-group"
                >
                  <label
                    htmlFor="password"
                  >
                    Password
                  </label>
                  <input
                    className="form-control"
                    name="password"
                    required={true}
                    type="password"
                  />
                </div>
                <button
                  className="btn btn-primary"
                  disabled={false}
                  type="submit"
                >
                  Login
                </button>
              </form>
              <br />
              <Link
                className="btn btn-link btn-sm"
                replace={false}
                to="/auth/create-account"
              />
              <Link
                className="btn btn-link btn-sm"
                replace={false}
                to="/auth/forgot-password"
              />
              <Link
                className="btn btn-link btn-sm"
                replace={false}
                to="/auth/reset-password"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</Login>
`;
